feat(curriculum): Added 20 new questions for the  'Semantic HTML' quiz (#59207)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/quiz-semantic-html/66ed903cf45ce3ece4053ebe.md

#### --answer--

Use the `i` element with the appropriate lang attribute to indicate that the text is in a different language.

### --question--

#### --text--

What does it mean when an HTML element is considered "non-semantic"?

#### --distractors--

It automatically applies default browser styles.

---

It provides no information about the meaning or purpose of its content.

---

It is only used within forms and interactive elements.

#### --answer--

It means the element does not convey any inherent meaning about its content; it is used solely for grouping or styling.

### --question--

#### --text--

What is the role of the `rt` element within a ruby annotation?

#### --distractors--

It encloses the base text that is being annotated

---

It provides fallback text if ruby annotations are not supported

---

It emphazises the wrapped text

#### --answer--

It specifies the annotation text (such as pronunciation or meaning) that appears above or below the base text in a ruby annotation

### --question--

#### --text--

Which of the following represents a proper way of defining a foreign language phrase?

#### --distractors--

```html

>   <i lng="fr">je ne sais quoi<</i>

Author: That was an accident.. It is corrected.

>   <i lng="fr">je ne sais quoi<</i>

Author: I hope this time there is no mistakes!

### --question--

#### --text--

How does semantic HTML improve search engine optimization (SEO)?

#### --distractors--

By adding hidden keywords to the page

---

By automatically generating meta tags

---

By reducing the amount of HTML code on the page

#### --answer--

By providing a meaningful structure that helps search engines understand the content better

### --question--

#### --text--

How does the `code` element differ from the `pre` element in representing code?

#### --distractors--

`code` automatically preserves whitespace while `pre` does not

---

`pre` is used for inline code snippets, whereas `code` is block-level

---

`code` applies syntax highlighting, unlike `pre`

#### --answer--

`code` marks a fragment of code inline, while `pre` preserves formatting and whitespace for multi-line code blocks.

### --question--

#### --text--

Which practice is recommended when marking up a foreign phrase within an English text?

#### --distractors--

Wrap the phrase in a `strong` element to denote importance.

---

Use the `i` element without any language attribute.

---

Enclose the phrase in a `span` element with inline styles.

#### --answer--

> Use the `i` element with the appropriate lang attribute to indicate that the text is in a different language.

Reviewer: Can you cite a reference for why you should use the `i` element?

Author: Of course! I used the content from the file [671a83934b61f64cefe87a61.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/25-front-end-development/review-semantic-html/671a83934b61f64cefe87a61.md) from line 31 to 38 to form this answer.

Reviewer: @vukedd Thank you for that. Is see why you have this question and the particular answer now. At first I read this as saying that you should always use the `i` element for marking up foreign language text, which is not the case. But you used "foreign phrase" which I think is in the spirit of using the `i` element. Really, it's a small technical point and doesn't make much of a difference since screen readers aren't going to treat the `i` element any differently than if you used a `span` here. But I'm OK keeping this as is.

Author: Glad to hear that!

### --question--

#### --text--

What does it mean when an HTML element is considered "non-semantic"?

#### --distractors--

It automatically applies default browser styles.

---

It provides no information about the meaning or purpose of its content.

---

It is only used within forms and interactive elements.

#### --answer--

It means the element does not convey any inherent meaning about its content; it is used solely for grouping or styling.

### --question--

#### --text--

What is the role of the `rt` element within a ruby annotation?

#### --distractors--

It encloses the base text that is being annotated

---

It provides fallback text if ruby annotations are not supported

---

> It emphazises the wrapped text

Reviewer 2: ```suggestion
It emphasizes the wrapped text
```

#### --answer--

It specifies the annotation text (such as pronunciation or meaning) that appears above or below the base text in a ruby annotation.

### --question--

#### --text--

Which of the following represents a proper way of defining a foreign language phrase?

#### --distractors--

```html
  <i lng="fr">je ne sais quoi</i>
```

---

```html
  <i language="fr">je ne sais quoi</i>
```

---

```html
  <i lg="fr">je ne sais quoi</i>
```

#### --answer--

```html
  <i lang="fr">je ne sais quoi</i>
```

### --question--

#### --text--

> Which element will be used to display text that has strong importance?

Reviewer 3: ```suggestion
Which element is used to display text that has strong importance?
```

#### --text--

Which element is used to annotate spelling errors?

#### --distractors--

`strong`

---

`em`

---

`s`

#### --answer--

`u`

### --question--

#### --text--

Which element would be useful for marking a phrase from another language?

#### --distractors--

`em`

---

`u`

---

`s`

#### --answer--

`i`

### --question--

#### --text--

What would be a common use case for the `sup` element?

#### --distractors--

To represent a phrase from another language.

---

To represent contact information.

---

To represent a section that is quoted from another source.

#### --answer--

To represent exponents, superior lettering and ordinal numbers.

### --question--

#### --text--

Why might a developer choose a semantic element over a generic `div` element?

#### --distractors--

Semantic elements load faster than `div` elements.

---

Semantic elements inherently provide meaning, which improves readability and accessibility.

---

Generic `div` elements are not supported on mobile devices.

#### --answer--

Because semantic elements add meaning to the code, making it more understandable for both developers and assistive technologies.

### --question--

#### --text--

> What is the purpose of the `footer` element in an HTML document?

Reviewer 3: This question should be replaced as the topic was covered in the Basic HTML section.
